db/commitlog/commitlog.cc

```diff
@@ -1593,6 +1621,15 @@ void db::commitlog::segment_manager::add_file_to_close(file f) {
     _files_to_close.emplace_back(std::move(f));
 }
 
+future<> db::commitlog::segment_manager::delete_file(const sstring& filename) {
+    return seastar::file_size(filename).then([this, filename](uint64_t size) {
+        // optimistic. because we don't want fails to prevent using a new segment
+        totals.total_size_on_disk -= size; 
+        clogger.debug("Deleting segment file {}", filename);
+        return commit_io_check(&seastar::remove_file, filename);
```

Comment:
Can this fail after we decreased total_size_on_disk above?

Reply:
Agree, should adjust after remove is suceessful (note that we'll take the node down if remove fails, but still).